docs/mp/openapi-geneator.adoc

```diff
@@ -0,0 +1,220 @@
+
+== Maven Coordinates [[Maven-Coordinates]]
+
+Core:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.openapitools</groupId>
+    <artifactId>openapi-generator</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+See the different versions of the openapi-generator artifact available on maven central.
+
+Cli:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.openapitools</groupId>
+    <artifactId>openapi-generator-cli</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+See the different versions of the openapi-generator-cli artifact available on maven central.
+
+Maven plugin:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.openapitools</groupId>
+    <artifactId>openapi-generator-maven-plugin</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+== Configuration
+
+To customize generated project different options can be used.
+Options for OpenAPI Generator are the same whether you're using the Maven/Gradle Plugins, CLI, or Online generation.
+
+General Configuration parameters
+
+At a minimum generation of the new Helidon MP project requires:
+
+[cols="1,1"]
+|===
+|Option | Description
+
+|inputSpec
+|OpenAPI Spec file path
+|generatorName
+|target generator name
+|library
+|library template (sub-template)
+|===
+
+The whole list of the general configuration parameters for OpenAPI Generator is available on link:https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin[openapi-generator-maven-plugin] page.
+
+Predefined values for required options for Helidon MP Generators :
+
+[cols="1,1,1"]
+|===
+|Option |Value |Description
+
+|generatorName
+|java-helidon-server
+|generator name for Helidon MP Server
+
+|generatorName
+|java-helidon-client
+|generator name for Helidon MP Client
+
+|library
+|mp
+|used library for Helidon MP Server/Client
+|===
+
+OpenAPI Generators for Helidon MP support such specific options :
+
+[cols="1,1,1,1,1"]
+|===
+|Option |Description |Values |Default |Notes
+
+|helidonVersion
+|Version of Helidon
+|
+|2.5.2
+|
+
+|fullProject
+|Generate API files only
+|true/false
+|false
+| API files : API Interfaces or abstract classes, models (files we do not expect the end user to modify)
+
+|serializationLibrary
+|Library to use for serializing JSON
+|jsonb, jackson
+|jackson
+|
+|===
+
+== Usage
+
+CLI, Maven/Gradle Plugins, or Online generation can be used to generate Helidon MP project.
+
+To generate a project using CLI at first `openapi-generator-cli.jar` or `Launcher Script` should be downloaded (details can be found in link:https://github.com/OpenAPITools/openapi-generator#1---installation[OpenAPI Generator GitHub Repository])
+
+To generate Helidon MP server stubs for petstore.yaml (OpenApi Document) using `openapi-generator-cli.jar`, please run the following :
```

Comment:
Typically, users run the generator from a directory _other than_ where the generator itself resides. Rather than navigating to that directory to run the generator, users would invoke the generator by specifying a fuller file spec to the generator itself, from anywhere in the file system.